### ISSUE TITLE
Question regarding unmarshalReflect

### ISSUE THREAD

**Issue by alexkohler** [guregu/dynamo#124]
Hi there, I'm currently running into an issue that I think may potentially be a bug, but wanted to get some background on the guarantees of `unmarshalReflect` (and the general premise of how this library performs unmarshaling). Inside of of the `unmarshalReflect`'s `reflect.Array` case, we have the following:
```
	case reflect.Array:
		arr := reflect.New(rv.Type()).Elem()
		elemtype := arr.Type().Elem()
		switch {
		case av.B != nil:
			for i, b := range av.B {
				arr.Index(i).Set(reflect.ValueOf(b))
			}
			rv.Set(arr)
			return nil
```

How are you guaranteeing that `len(av.B)` is equal to (or less than) `arr.Len()`?

**Comment by guregu** [comment]
Hello
Looks like a bug to me 😅
After messing around with encoding/json a bit, it seems to discard values that won’t fit into the array
We should probably match that behavior
Although I think an error message might be more useful 🤔

**Comment by alexkohler** [comment]
Either approach seems better than a panic :smile: It might be more idiomatic to match encoding/json's behavior, but the behavior should be explicitly documented either way.

**Comment by guregu** [comment]
I released v1.7.1 which fixes this. I decided to return an error when the array is too small. I imagine most of the time unmarshaling into a too-small array is a mistake instead of intended behavior, and silently truncating data is scary. I can always change this later if there's demand for it.

I haven't updated the docs yet but I'd like to have more explicit info about how each type is handled, I added this to #45.

Thanks for the bug report!
